tracer/src/Datadog.Trace/PlatformHelpers/AspNetCoreHttpRequestHandler.cs

```diff
@@ -118,8 +120,12 @@ public Scope StartAspNetCorePipelineScope(Tracer tracer, HttpContext httpContext
             }
 
             var scope = tracer.StartActiveInternal(_requestInOperationName, propagatedContext, tags: tags);
-
-            scope.Span.DecorateWebServerSpan(resourceName, httpMethod, host, url, tags, tagsFromHeaders);
+            scope.Span.DecorateWebServerSpan(resourceName, httpMethod, host, url, userAgent, tags, tagsFromHeaders);
+            var peerIp = new Headers.Ip.IpInfo(httpContext.Connection.RemoteIpAddress?.ToString(), httpContext.Connection.RemotePort);
```

Comment:
no worries if the first param is null I assume? Like no exception can be thrown while processing this ip?

Reply:
nope, this ipaddress is just a fallback :)